src/scripts/main.js

```diff
@@ -1 +1,72 @@
 'use strict';
+
+const body = document.querySelector('body');
+
+const resolver1 = (resolve, reject) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0) {
+      resolve();
+    }
+  });
+
+  setTimeout(() => {
+    reject();
+  }, 3000);
+};
+
+const resolver2 = (resolve) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve();
+    }
+  });
+};
+
+const resolver3 = (resolve) => {
+  document.addEventListener('mousedown', (e1) => {
+    document.addEventListener('mousedown', (e2) => {
+      if ((e1.button === 0 && e2.button === 2)
+      || (e2.button === 0 && e1.button === 2)) {
+        resolve();
+      }
+    });
+  });
```

Comment:
Don't use nested listener. Create 2 variables instead of and change it values after right and left click. So when you variables show you that there were clicks on both buttons resolve third promise